runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/stream/resolver/EnvironmentResolverSpi.java

```diff
@@ -0,0 +1,28 @@
+package io.aklivity.zilla.runtime.engine.internal.stream.resolver;
+
+import java.util.Optional;
+import java.util.function.UnaryOperator;
+import java.util.regex.Pattern;
+
+import io.aklivity.zilla.runtime.engine.resolver.ExpressionResolverSpi;
+
+public class EnvironmentResolverSpi implements ExpressionResolverSpi
+{
+
+    private static final Pattern MUSTACHE_PATTERN = Pattern.compile("\\$\\{\\{\\s*env\\.([^\\s\\}]*)\\s*\\}\\}");
```

Comment:
Rename to `EXPRESSION_PATTERN` instead.

Also, this can be moved to the `ExpressionResolver` itself, match the pattern, extract the context, then use the appropriate `ExpressionResolverSpi` to resolve the value.

Each `ExpressionResolverSpi` implementation should be as simple as possible to avoid duplication when we extend to support more contexts.